packages/backend/src/analytics/LightdashAnalytics.ts

```diff
@@ -1564,4 +1561,33 @@ export class LightdashAnalytics extends Analytics {
             context: { ...this.lightdashContext }, // NOTE: spread because rudderstack manipulates arg
         });
     }
+
+    /**
+     * Track events from an account. This lets us centralize common properties that are found in req.account.
+     */
+    trackAccount<T extends BaseTrack>(
+        account: Account,
+        basePayload: TypedEvent | UntypedEvent<T>,
+    ) {
+        const payload = {
+            ...basePayload,
+        };
+
+        if (!payload.properties) {
+            payload.properties = {};
+        }
+
+        if (account.authentication.type === 'session') {
+            payload.userId = account.user.id;
+        } else {
+            payload.anonymousId = 'embed';
```

Comment:
is this safe to assume all "non session" will be used for embed ? will we have to remember this when we start using jwt tokens for other purposes ?

Reply:
Ah that's a good call. I think I landed on the right type name to user for the user: `known` which implies this is a user in the users table. A better check here will be `account.user.type === 'known'`. That's what we care about for this because `userId` if used will be joined to the users table in dbt.